net/sing-box/Makefile

```diff
@@ -0,0 +1,134 @@
+
+PKG_BUILD_DEPENDS:=golang/host
+PKG_BUILD_PARALLEL:=1
+PKG_USE_MIPS16:=0
+
+GO_PKG:=github.com/sagernet/sing-box
+GO_PKG_BUILD_PKG:=$(GO_PKG)/cmd/sing-box
+
+GO_PKG_LDFLAGS:=-s -w
+GO_PKG_LDFLAGS_X:= \
+	$(GO_PKG).Version=$(PKG_VERSION)
+
+include $(INCLUDE_DIR)/package.mk
+include ../../lang/golang/golang-package.mk
+
+define Package/sing-box
+  TITLE:=The universal proxy platform
+  SECTION:=net
+  CATEGORY:=Network
+  URL:=https://sing-box.sagernet.org
+  DEPENDS:=$(GO_ARCH_DEPENDS)
+  USERID:=sing-box=5566:sing-box=5566
+endef
+
+define Package/sing-box/description
+  Sing-box is a universal proxy platform which supports hysteria, SOCKS, Shadowsocks,
+  ShadowsocksR, ShadowTLS, Tor, trojan, VLess, VMess, WireGuard and so on.
+endef
+
+define Package/sing-box/config
+	menu "Configure Options"
+		depends on PACKAGE_sing-box
+
+		config SINGBOX_WITH_QUIC
+			bool "Build with QUIC support"
+			default y
+
+	 config SINGBOX_WITH_GRPC
+			bool "Build with standard gRPC support"
+			default n
+
+	 config SINGBOX_WITH_WIREGUARD
+			bool "Build with WireGuard support"
+			default y
+
+	 config SINGBOX_WITH_SHADOWSOCKSR
+			bool "Build with ShadowsocksR support"
+			default n
+
+	 config SINGBOX_WITH_ECH
+			bool "Build with TLS ECH extension support for TLS outbound"
+			default n
+
+	 config SINGBOX_WITH_UTLS
+			bool "Build with uTLS support for TLS outbound"
+			default y
+
+	 config SINGBOX_WITH_ACME
+			bool "Build with ACME TLS certificate issuer support"
+			default n
+
+	 config SINGBOX_WITH_CLASH_API
+			bool "Build with Clash API support"
+			default y
+
+	 config SINGBOX_WITH_V2RAY_API
+			bool "Build with V2Ray API support"
+			default n
+
+	 config SINGBOX_WITH_GVISOR
+			bool "Build with gVisor support"
+			default y
+
+	 config SINGBOX_WITH_EMBEDDED_TOR
+			bool "Build with embedded Tor support"
+			default n
+
+	 config SINGBOX_WITH_LWIP
+			bool "Build with LWIP Tun stack support"
+			default n
+	endmenu
+endef
+
+GO_PKG_TAGS:=$(subst $(space),$(comma),$(strip \
+	 $(if $(CONFIG_SINGBOX_WITH_QUIC),with_quic) \
+	 $(if $(CONFIG_SINGBOX_WITH_GRPC),with_grpc) \
+	 $(if $(CONFIG_SINGBOX_WITH_WIREGUARD),with_wireguard) \
+	 $(if $(CONFIG_SINGBOX_WITH_SHADOWSOCKSR),with_shadowsocksr) \
+	 $(if $(CONFIG_SINGBOX_WITH_ECH),with_ech) \
+	 $(if $(CONFIG_SINGBOX_WITH_UTLS),with_utls) \
+	 $(if $(CONFIG_SINGBOX_WITH_ACME),with_acme) \
+	 $(if $(CONFIG_SINGBOX_WITH_CLASH_API),with_clash_api) \
+	 $(if $(CONFIG_SINGBOX_WITH_V2RAY_API),with_v2ray_api) \
+	 $(if $(CONFIG_SINGBOX_WITH_GVISOR),with_gvisor) \
+	 $(if $(CONFIG_SINGBOX_WITH_EMBEDDED_TOR),with_embedded_tor) \
+	 $(if $(CONFIG_SINGBOX_WITH_LWIP),with_lwip) \
+	 ))
+
+define Package/sing-box/conffiles
+/etc/config/sing-box
+/etc/sing-box/
+endef
+
+define Build/Install
+	$(call GoPackage/Package/Install/Bin,$(PKG_INSTALL_DIR))
+endef
+
+define Package/sing-box/install
+	$(INSTALL_DIR) $(1)/usr/bin/
+	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/sing-box $(1)/usr/bin/sing-box
```

Comment:
```suggestion
define Package/sing-box/install
	$(call GoPackage/Package/Install/Bin,$(1))
```

Reply:
I don't see multiple binaries / packages in sing-box package, only `/usr/bin/sing-box` exists.

Here is golang, so don't mention cmake or other stuffs. Also, in the most cases, you won't need to handle `Build/Install` manually.
https://github.com/openwrt/packages/blob/582fc64a948f224fbcbccf1fe1616a2841254683/lang/golang/golang-package.mk#L309-L313

Also, please define `PKG_CONFIG_DEPENDS` for your configs.
fyi https://github.com/immortalwrt/packages/blob/master/net/sing-box/Makefile#L19

Reply:
> I think, this is more flexible and can work for both cases.

So, you are assuming something won't happen.

> Taking cmake and meson as an example, I want to say that the $(PKG_INSTALL_DIR) is where stuffs generated by the build system should be installed first.

What about take a look at how other contributors did: https://github.com/search?q=repo:openwrt/packages+golang&type=code

`Build/Install` is totally unused by any golang package, also unnecessary.

If you insist on using `PKG_INSTALL_DIR` somehow, please define it under `Package/install`:
https://github.com/openwrt/packages/blob/582fc64a948f224fbcbccf1fe1616a2841254683/net/v2raya/Makefile#L78-L81